DuckDuckGo/SyncSettingsViewController.swift

```diff
@@ -132,7 +131,12 @@ class SyncSettingsViewController: UIHostingController<SyncSettingsView> {
     }
 
     func dismissPresentedViewController() {
-        navigationController?.topViewController?.dismiss(animated: true)
+        if let presentedViewController = navigationController?.presentedViewController {
+            if !(presentedViewController is UIHostingController<SyncSettingsView>) {
+                presentedViewController.dismiss(animated: true, completion: nil)
+                endConnectMode()
+            }
+        }
```

Comment:
This would be better as a guard.  The reason being this entire function depends on these two conditions being true - and bonus it'll reduce some indenting.

```suggestion
        guard let presentedViewController = navigationController?.presentedViewController,
              !(presentedViewController is UIHostingController<SyncSettingsView>) else { return }
        presentedViewController.dismiss(animated: true, completion: nil)
        endConnectMode()
```